src/wasm-ast-checker.c

```diff
@@ -1059,12 +1059,11 @@ static void check_elem_segments(Context* ctx, const WasmModule* module) {
             offset, last_end);
       }
 
-      uint64_t max =
-          table->elem_limits.has_max ? table->elem_limits.initial : UINT32_MAX;
-
-      if (offset + elem_segment->vars.size > max) {
+      uint64_t max = table->elem_limits.initial;
+      if ((uint64_t)offset + elem_segment->vars.size > max) {
         print_error(ctx, CHECK_STYLE_FULL, &field->loc,
-                    "segment ends past the end of table max size (%" PRIu64 ")",
+                    "segment ends past the end of table initial size (%" PRIu64
```

Comment:
The grammar seems a little off to me here.  How about just `segment ends past the end of the table`?   `segment ends outside the initial table table size`?